pkg/filter/filter.go

```diff
@@ -113,9 +113,14 @@ func NewFilter(cfg *config.FilterConfig, tz string, caseSensitive bool, forceRep
 
 // ShouldIgnoreDML checks if a DML event should be ignore by conditions below:
 // 0. By startTs.
-// 1. By table name.
-// 2. By type.
+// 1. By eligibility of the table.
+// 2. By table name.
+// 3. By type.
 func (f *filter) ShouldIgnoreDML(dmlType common.RowType, preRow, row chunk.Row, tableInfo *common.TableInfo, startTs uint64) (bool, error) {
+	if !f.isEligible(tableInfo) {
```

Comment:
This check is unnecessary. If `forceReplicate=false`, ineligible tables should not be scheduled for replication in the first place.
Performing the check here would cause confusion.

Reply:
```go
func (f *filter) isEligible(tableInfo *common.TableInfo) bool {
	if tableInfo == nil {
		// This should happen in test only
		return true
	}
	return tableInfo.IsEligible(f.forceReplicate)
}
```

This is the implementation of the isEligible function. Here, the forceReplicate configuration is passed to TableInfo.IsEligible. Because the forceReplicate configuration is stored in the filter, the isEligible function needs to be called, and this is the earliest place where DML can be filtered.